tools/bin/ariaDownload.py

```diff
@@ -188,6 +196,74 @@ def _fmt_dst(self):
             count += 1
         return dst
 
+    def make_nc_cookie(self):
+       import netrc, base64
+       from urllib.request import build_opener, Request
+       from urllib.request import HTTPHandler, HTTPSHandler, HTTPCookieProcessor
+       from urllib.error import HTTPError, URLError
+       from http.cookiejar import MozillaCookieJar
+
+       cookie_jar_path = op.join(op.expanduser('~'), '.bulk_download_cookiejar.txt')
+       asf_urs4        = {'url': 'https://urs.earthdata.nasa.gov/oauth/authorize',
+                         'client': 'BO_n7nTIlMljdvU6kRRB3g',
+                         'redir': 'https://auth.asf.alaska.edu/login'}
+       if self.inps.user is not None and self.inps.passw is not None:
+           user  = self.inps.user
+           passw = self.inps.passw
+       elif op.exists(op.join(op.expanduser('~'), '.netrc')):
+           # log.info('Obtaining user/pass from .netrc')
+           user, _,  passw = netrc.netrc().authenticators('urs.earthdata.nasa.gov')
+       else:
+           # resort to ASF credential checks (will prompt for input if can't find the cookiejar)
+           return None
+
+       # Build URS4 Cookie request
+       auth_cookie_url = f"{asf_urs4['url']}?client_id={asf_urs4['client']}&redirect_uri={asf_urs4['redir']}&response_type=code&state="
+       user_pass = base64.b64encode(bytes(f'{user}:{passw}', 'utf-8')).decode('utf-8')
+
+       # Authenticate against URS, grab all the cookies
+       cookie_jar = MozillaCookieJar()
+       opener     = build_opener(HTTPCookieProcessor(cookie_jar), HTTPHandler(), HTTPSHandler(**{}))
+       request    = Request(auth_cookie_url, headers={'Authorization': f'Basic {user_pass}'})
+
+       # Watch out cookie rejection!
+       try:
+          response = opener.open(request)
+       except HTTPError as e:
+          if "WWW-Authenticate" in e.headers and "Please enter your Earthdata Login credentials" in e.headers["WWW-Authenticate"]:
+             print (" > Username and Password combo was not successful. Please try again.")
+             return False
+          else:
+             # If an error happens here, the user most likely has not confirmed EULA.
+             print ('\nThere was an error obtaining a download cookie')
+             print ('Most likely you lack permission to download data from the ASF Datapool.')
+             print ('\n\nNew users: you must first log into Vertex and accept the EULA. In addition, your Study Area must be set at Earthdata https://urs.earthdata.nasa.gov')
+             os.sys.exit(-1)
+
+       except URLError as e:
```

Comment:
Codacy found an issue: [local variable 'e' is assigned to but never used (F841)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=5918918)